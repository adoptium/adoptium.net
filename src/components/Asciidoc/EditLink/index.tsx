import React from "react"
import Image from "next/image"
import { useTranslations } from "next-intl"
import { IoMdGitPullRequest } from "react-icons/io"

type Props = {
  relativePath?: string
}

const EditLink: React.FC<Props> = ({ relativePath }) => {
  const t = useTranslations("EditLink")
  if (!relativePath) {
    return null
  }

  const href = `https://github.com/adoptium/adoptium.net/edit/main/content/asciidoc-pages/${relativePath}`

  return (
    <div className="w-full backdrop-blur-sm bg-gradient-to-br from-[#20114D]/30 to-[#2A0845]/40 px-8 py-6 rounded-2xl border border-white/10 shadow-lg transform transition-all duration-300 hover:shadow-xl hover:border-white/20 flex flex-col md:flex-row justify-between items-center gap-6">
      <div className="flex items-center gap-5">
        <div className="relative w-16 h-16 md:w-20 md:h-20 flex-shrink-0">
          <div className="absolute inset-0 bg-gradient-to-tr from-pink-500 to-purple-600 rounded-full opacity-30 blur-md"></div>
          <Image
            fill
            className="relative w-full h-full p-3 bg-[#200D46]/60 border border-white/20 rounded-full shadow-md"
            src="/images/icons/edit.svg"
            alt="edit icon"
          />
        </div>
        <div className="flex flex-col gap-2">
          <h3 className="text-white text-xl md:text-2xl font-bold leading-tight">
            {t("help-us-make-these-docs-great")}
          </h3>
          <p className="text-white/90 text-sm md:text-base font-normal leading-relaxed max-w-lg">
            {t("all-adoptium-docs-are-open-source")}
          </p>
        </div>
      </div>
      <a href={href} target="_blank" rel="noopener noreferrer" className="flex-shrink-0">
        <button className="group relative px-6 py-3 overflow-hidden rounded-xl bg-gradient-to-r from-[#FF3366] to-[#9933FF] text-white text-base leading-6 font-medium shadow-md transition-all duration-300 hover:shadow-lg hover:scale-105">
          <span className="relative z-10 flex items-center gap-2">
            <IoMdGitPullRequest className="h-5 w-5" />
            {t("edit-this-page")}
          </span>
          <div className="absolute inset-0 w-full h-full bg-gradient-to-r from-pink-600 to-purple-700 opacity-0 group-hover:opacity-100 transition-opacity duration-300"></div>
        </button>
      </a>
    </div>
  )
}

export default EditLink
