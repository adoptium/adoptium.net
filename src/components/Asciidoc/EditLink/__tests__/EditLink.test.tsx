import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it } from "vitest"
import EditLink from ".."

describe("EditLink component", () => {
  it("renders correctly", () => {
    const { container } = render(<EditLink relativePath="/fake/path.adoc" />)
    expect(container).toMatchSnapshot()
  })

  it("does not render without a path", () => {
    const { container } = render(<EditLink relativePath="" />)
    expect(container).toMatchSnapshot()
  })
})
