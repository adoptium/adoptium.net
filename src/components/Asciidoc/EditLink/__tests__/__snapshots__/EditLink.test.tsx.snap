// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditLink component > does not render without a path 1`] = `<div />`;

exports[`EditLink component > renders correctly 1`] = `
<div>
  <div
    class="w-full backdrop-blur-sm bg-gradient-to-br from-[#20114D]/30 to-[#2A0845]/40 px-8 py-6 rounded-2xl border border-white/10 shadow-lg transform transition-all duration-300 hover:shadow-xl hover:border-white/20 flex flex-col md:flex-row justify-between items-center gap-6"
  >
    <div
      class="flex items-center gap-5"
    >
      <div
        class="relative w-16 h-16 md:w-20 md:h-20 flex-shrink-0"
      >
        <div
          class="absolute inset-0 bg-gradient-to-tr from-pink-500 to-purple-600 rounded-full opacity-30 blur-md"
        />
        <img
          alt="edit icon"
          class="relative w-full h-full p-3 bg-[#200D46]/60 border border-white/20 rounded-full shadow-md"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          src="/images/icons/edit.svg"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <div
        class="flex flex-col gap-2"
      >
        <h3
          class="text-white text-xl md:text-2xl font-bold leading-tight"
        >
          Text
        </h3>
        <p
          class="text-white/90 text-sm md:text-base font-normal leading-relaxed max-w-lg"
        >
          Text
        </p>
      </div>
    </div>
    <a
      class="flex-shrink-0"
      href="https://github.com/adoptium/adoptium.net/edit/main/content/asciidoc-pages//fake/path.adoc"
      rel="noopener noreferrer"
      target="_blank"
    >
      <button
        class="group relative px-6 py-3 overflow-hidden rounded-xl bg-gradient-to-r from-[#FF3366] to-[#9933FF] text-white text-base leading-6 font-medium shadow-md transition-all duration-300 hover:shadow-lg hover:scale-105"
      >
        <span
          class="relative z-10 flex items-center gap-2"
        >
          <svg
            class="h-5 w-5"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 376.6V206.8c0-46.4-3.7-70.8-22.1-94C374 87.7 348.9 77 310 77h-22V16l-96 96 96 96v-65h22c24.6 0 29 3.6 33.8 9.6 5.5 6.9 8.2 19.1 8.2 54.2v169.8c-19.1 11.1-32 31.7-32 55.4 0 35.3 28.7 64 64 64s64-28.7 64-64c0-23.7-12.9-44.3-32-55.4zM384 472c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zM128 48c-35.3 0-64 28.7-64 64 0 23.7 12.9 44.3 32 55.4v209.2c-19.1 11.1-32 31.7-32 55.4 0 35.3 28.7 64 64 64s64-28.7 64-64c0-23.7-12.9-44.3-32-55.4V167.4c19.1-11.1 32-31.7 32-55.4 0-35.3-28.7-64-64-64zm0 424c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm0-320c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40z"
            />
          </svg>
          Text
        </span>
        <div
          class="absolute inset-0 w-full h-full bg-gradient-to-r from-pink-600 to-purple-700 opacity-0 group-hover:opacity-100 transition-opacity duration-300"
        />
      </button>
    </a>
  </div>
</div>
`;
